### ISSUE TITLE
de-TLD, etc. need to redirect to wheelmap.org

### ISSUE THREAD

**Issue by thermann78** [sozialhelden/wheelmap#51]
Error in wheelmap_production.conf:

server {
  listen 80;
  server_name wheelmap.de wheelmaps.de www.wheelmap.de www.wheelmaps.de;
  rewrite_log on;
  rewrite ^(.*)$ http://wheelmap.org/de$1 permanent;
}

There is no dir named /de, so all requests will be forwarded by WP to a blogpost.
Please point to wheelmap.org/$1.

**Comment by christoph-buente** [comment]
This is actually just a locallized forwarding, cause we used to use subdomains for locale switching in the rails app. It is non relevant for wordpress.

**Comment by thermann78** [comment]
Just tryout: http://wheelmap.org/de
There is a general error in the nginx config: the correct forward string should look like 
# redirect japanese subdomain
server {
  listen 80;
  server_name jp.wheelmap.org;
  rewrite_log on;
  rewrite ^(.)$ http://wheelmap.org/ja/map permanent;
}

Note: this concerns many redir rules (except english, we actually have an /en/-subdirectory.

**Comment by lennerd** [comment]
Where are this nginx config placed? And how can we access them?

**Comment by thermann78** [comment]
/etc/nginx/sites-available

**Comment by holgerd** [comment]
@Xylakant can I give this to you? 
www.wheelmap.de and wheelmap.de should simply 301 redirect to http://wheelmap.org

**Comment by lennerd** [comment]
So, this also includes to fix the wrong proxy for http://wheelmap.org/de to the blog.

**Comment by holgerd** [comment]
This is still open: http://wheelmap.org/de and www.wheelmap.de still redirect incorrectly to https://community.wheelmap.org/der-social-day-2015-mit-immobilienscout24/
@Xylakant can you look into this until the end of next week?

**Comment by Xylakant** [comment]
we seem to have various rewrites that have errors as well, for example fr.wheelmap.org which redirects to a 404 page. what's the proper redirect target for all language specific domaings? the map?

**Comment by thermann78** [comment]
Yes. http://wheelmap.org/fr/map would be correct.

**Comment by Xylakant** [comment]
what should happen with  http://wheelmap.org/fr? making redirects for all of those language versions is going to be problematic since we need to explicitly list all of them.

**Comment by Xylakant** [comment]
pointing to  wheelmap.org/$1 as proposed is certainly the wrong solution and will break existing links to wheelmap.de/map.

**Comment by holgerd** [comment]
I just found that the wheelmap.de domain is also aliased & redirected using the cpanel of planetary networks (our blog hoster):

![cpanel_-_aliase](https://cloud.githubusercontent.com/assets/218506/14977187/5349e36a-1112-11e6-8b6f-e0e495cddfb4.png)


![cpanel_-_umleitungen](https://cloud.githubusercontent.com/assets/218506/14977214/7789f24c-1112-11e6-989a-f862c6ba73f7.png)

This seems strange to me since I also read here that the redirect is done in the nginx config?

**Comment by Xylakant** [comment]
The redirect happens in nginx, the cpanel config has no effect as long as the domain points to cloudflare.

**Comment by holgerd** [comment]
My suggestions to finally close this issue:

1. www.wheelmap.org is our desired TLD, all other domains should redirect to it (wheelmap.de, wheelmap.com, wheelmap.net, accessforeverybody.com, accessforeverybody.net and the same with www.)
2. We do not support localized sub-domains anymore. since a long time actually. Instead of thinking of any complex rules, why not igonre fr.wheelmap.org etc. completely? Then it would correctly show as 404.
3. community.wheelmap.org stays the landing page for the blog content.

@Xylakant @thermann78 @lennerd @1000miles  makes sense?

**Comment by thermann78** [comment]
I don't think this makes more sense than correcting the existing config by adding /(lang)/map to the target URI -it is exactly the same effort, as @Xylakant has to edit the Nginx conf and either delete all these redirects or complete them - but this is your decision. I am just unsure if deleting these rules will cause any SEO probs.

Wheelmap.de is of course ruled in cPanel as Planetary is our domain registrar and used as nameserver. We would have to pay for additional TLDs in Cloudflare so our additional domains are registered and nameserved with Planetary and redirected to wheelmap.org where they come to Cloudflare.
As these additional domains aren't used frequently and without high traffic there is no need to take them to CF, so there are two options to call the Wheelmap:

1. Via Wheelmap.org -> Cloudflare -> Nginx on Spiderman -> Spiderman or blogserver.
2. Via Wheelmap.de or another TLD -> Planetary DNS -> Wheelmap.org -> (see 1.)

**Comment by holgerd** [comment]
@thermann78 why is the whelmap.de -> wheelmap.org redirect not working then?

**Comment by thermann78** [comment]
It is. The Nginx conf is called AFTER redirecting. Nginx conf is the final redirect that overrides all previous rules.
The general error is that in Nginx conf the redirect rule for wheelmap.de isn't wheelmap.org/de/map but wheelmap.org/de.
Just add /map to all country redirects (all but /en) and it works.

**Comment by holgerd** [comment]
@thermann78 in cPanel I can see that wheelmap.de is redirecting to http://www.wheelmap.org/ . If that worked correctly, then there is no Nginx redirect afterwards. When I type in http://www.wheelmap.org/ in my browser it works.

**Comment by thermann78** [comment]
Again: Request -> DNS-Lookup at nameserver -> IP-resolve -> redirect to server -> local configuration -> redirect to internal directory. 
Of course there is a Nginx redirect - internal redirects are the server's job.

**Comment by holgerd** [comment]
@thermann78 and I just discussed this issue. Result:

It seems that nginx is doing something smart when the referer ist wheelmap.de which leads to the error.

So I update my suggestions:

- www.wheelmap.org is our desired TLD, all other domains should redirect to it (wheelmap.de, wheelmap.com, wheelmap.net, accessforeverybody.com, accessforeverybody.net and the same with www.)
- We do not support localized sub-domains anymore. since a long time actually. Instead of thinking of any complex rules, why not ignore fr.wheelmap.org etc. completely? Then it would correctly show as 404.
- Also ignore wheelmap.de in nginx since it will be redirected 301 to wheelmap.org anyways.
- community.wheelmap.org stays the landing page for the blog content.

**Comment by 1000miles** [comment]
I checked the current state of the locale redirects on the wheelmap.org. 

**In case it may help, here is an overview:**

* de.wheelmap.org (german) => currently redirects to http://wheelmap.org/map
* en.wheelmap.org (english) => currently redirects to http://wheelmap.org/en
* it.wheelmap.org (italian) => currently redirects to http://wheelmap.org/it
* is.wheelmap.org (icelandic) => currently redirects to http://wheelmap.org/is/map
* el.wheelmap.org (greek) => currently redirects to https://community.wheelmap.org/el **(currently broken, Error 404)**
* es.wheelmap.org (spanish) => currently redirects to https://community.wheelmap.org/es **(currently broken, Error 404)**
* fr.wheelmap.org (french) => currently redirects to https://community.wheelmap.org/fr **(currently broken, Error 404)**
* pt-pt.wheelmap.org (portuguese) => currently redirects to http://wheelmap.org/pt_BR/map # why different here: pt-pt & pt_BR?
* ar.wheelmap.org (arab) => currently redirects to http://wheelmap.org/map # should point to wheelmap.org/ar/map
* bg.wheelmap.org (bulgarian) => currently redirects to http://wheelmap.org/map  # should point to wheelmap.org/bg/map
* jp.wheelmap.org (japanese) => currently redirects to http://wheelmap.org/ja/map  # why `ja` instead of `jp`?
* tlh.wheelmap.org (clingon) => currently redirects to http://wheelmap.org/tlh/map
* hu.wheelmap.org (hungarian) => currently redirects to http://wheelmap.org/map # should point to http://wheelmap.org/hu//map
* ko.wheelmap.org (korean) => currently redirects to http://wheelmap.org/map # should point to 
http://wheelmap.org/ko/map
* lv.wheelmap.org (latvian) => currently redirects to http://wheelmap.org/map # should point to 
http://wheelmap.org/lv/map
* pl.wheelmap.org (polish) => currently redirects to http://wheelmap.org/map # should point to 
http://wheelmap.org/pl/map
* pt_BR.wheelmap.org (brasilian-portuguese)=> currently redirects to http://wheelmap.org/map # should point to 
http://wheelmap.org/pt_BR/map
* ru.wheelmap.org (romanian) => currently redirects to http://wheelmap.org/map # should point to 
http://wheelmap.org/ru/map
* sk.wheelmap.org (slovak) => currently redirects to http://wheelmap.org/map # should point to 
http://wheelmap.org/sk/map 
* tr.wheelmap.org (turkish) => currently redirects to http://wheelmap.org/map # should point to 
http://wheelmap.org/tr/map
* zh_TW.wheelmap.org (taiwanese) => currently redirects to http://wheelmap.org/map # should point to 
http://wheelmap.org/zh_TW/map

**Comment by holgerd** [comment]
We plan to omit all language-specific redirects without replacing them. This will lead to 404 for all of them.

@thermann78 @1000miles 
So all which needs to be done is to redirect wheelmap.de and wheelmaps.de to wheelmap.org (let's not be smart about language switches here).

**Comment by holgerd** [comment]
@thermann78 maybe you can look up what other TLDs we have (com, net??) and redirect them all to .org?

**Comment by thermann78** [comment]
This is how the configuration is set in cPanel before the requests are handled by Nginx-Config:

![bildschirmfoto 2016-08-31 um 12 42 12](https://cloud.githubusercontent.com/assets/15140420/18125837/9b9ea672-6f78-11e6-9cd7-59b36fb8ab6b.JPG)

**Comment by holgerd** [comment]
This is imho how it should be:

Redirect these:
http://accessforeverybody.org
https://accessforeverybody.org (this goes to an external site right now!?)
http://wheelmap.com
https://wheelmap.com
http://wheelmap.de
https://wheelmap.de
http://wheelmap.net
https://wheelmap.net

with 301 to:
https://wheelmap.org

AND

http://www.accessforeverybody.org
https://www.accessforeverybody.org (this goes to an external site right now!?)
http://www.wheelmap.de
https://www.wheelmap.de
http://www.wheelmap.com
https://www.wheelmap.com
http://www.wheelmap.net
https://www.wheelmap.net
http://www.wheelmap.org

with 301 to:
https://wheelmap.org

**Comment by holgerd** [comment]
this is where this issue stands imho:

- [ ] omit language code subdomains like it.wheelmap etc
- [ ] redirect TLDs

**Comment by Xylakant** [comment]
@holgerd This will be slightly tricky: In the ideal case, the domains all point to the same ip as wheelmap.org does. The IP for wheelmap.org may change at some point (it's a floating IP on the sys11 side, so it may change on short notice). However, the bare domain (wheelmap.de) cannot reference another domain, (that would be a CNAME DNS record) but needs to specify an IP address (A record). Some DNS providers allow faking a CNAME on a bare domain (They literally retrieve the A record of the domain you're pointing to and then generate the A record on the fly). I'm a bit skeptical if planetary networks supports that. Cloudflare does support it afaik, but adding a CF account for each site would be expensive. DNSimple does, so I'd recommend moving the domains DNS at least there. We'd have to check if planetary networks allows us to use different nameservers for domains registered there.

redirecting https will be even more of an issue: we'd need a cert that contains all those domain names and that would again require us to have a cloudflare plan for each of them. I'd just not support https for anything that's not wheelmap.org

**Comment by Xylakant** [comment]
I enabled "always use HTTPS" for wheelmap.org.

**Comment by Xylakant** [comment]
I added a strikethrough to the domains that can't be done without a dedicated cloudflare plan or dedicated certificates at least.

**Comment by Xylakant** [comment]
Taken from @1000miles list

- [ ] de.wheelmap.org (german) => currently redirects to http://wheelmap.org/map # should point to https://wheelmap.org/map
- [ ] en.wheelmap.org (english) => currently redirects to http://wheelmap.org/en/map # should point to https://wheelmap.org/map
- [ ] it.wheelmap.org (italian) => currently redirects to http://wheelmap.org/it/map # should point to https://wheelmap.org/map
- [ ] is.wheelmap.org (icelandic) => currently redirects to http://wheelmap.org/is/map # should point to https://wheelmap.org/map
- [ ] el.wheelmap.org (greek) => currently redirects to https://community.wheelmap.org/el **(currently broken, Error 404)** # should point to https://wheelmap.org/map
- [ ] es.wheelmap.org (spanish) => currently redirects to https://community.wheelmap.org/es **(currently broken, Error 404)** # should point to https://wheelmap.org/map
- [ ] fr.wheelmap.org (french) => currently redirects to https://community.wheelmap.org/fr **(currently broken, Error 404)** # should point to https://wheelmap.org/map
- [ ] pt-pt.wheelmap.org (portuguese) => currently redirects to http://wheelmap.org/pt_BR/map # why different here: pt-pt & pt_BR? # should point to https://wheelmap.org/map
- [ ] ar.wheelmap.org (arab) => currently redirects to http://wheelmap.org/map # should point to https://wheelmap.org/map
- [ ] bg.wheelmap.org (bulgarian) => currently redirects to http://wheelmap.org/map  # should point to # should point to https://wheelmap.org/map
- [ ] jp.wheelmap.org (japanese) => currently redirects to http://wheelmap.org/ja/map  # why `ja` instead of `jp`? # should point to https://wheelmap.org/map
- [ ] tlh.wheelmap.org (clingon) => currently redirects to http://wheelmap.org/tlh/map # should point to https://wheelmap.org/map
- [ ] hu.wheelmap.org (hungarian) => currently redirects to http://wheelmap.org/map # should point to https://wheelmap.org/map
- [ ] ko.wheelmap.org (korean) => currently redirects to http://wheelmap.org/map # should point to https://wheelmap.org/map
- [ ] lv.wheelmap.org (latvian) => currently redirects to http://wheelmap.org/map # should point to https://wheelmap.org/map
- [ ] pl.wheelmap.org (polish) => currently redirects to http://wheelmap.org/map # should point to https://wheelmap.org/map
- [ ] pt_BR.wheelmap.org (brasilian-portuguese)=> currently redirects to http://wheelmap.org/map # should point to https://wheelmap.org/map
- [ ] ru.wheelmap.org (romanian) => currently redirects to http://wheelmap.org/map # should point to https://wheelmap.org/map
- [ ] sk.wheelmap.org (slovak) => currently redirects to http://wheelmap.org/map # should point to https://wheelmap.org/map 
- [ ] tr.wheelmap.org (turkish) => currently redirects to http://wheelmap.org/map # should point to https://wheelmap.org/map
- [ ] zh_TW.wheelmap.org (taiwanese) => currently redirects to http://wheelmap.org/map # should point to https://wheelmap.org/map

**Comment by Xylakant** [comment]
country specific subdomains are done. please report all weirdness to me.

**Comment by Xylakant** [comment]
@holgerd are you certain you want to mark this as done? according to https://github.com/sozialhelden/wheelmap/issues/51#issuecomment-244712636 there's still work left to do (though all of it something we can't do)

**Comment by thermann78** [comment]
@holgerd I agree with @Xylakant - the initial problem isn't solved at all, just have a click on http://wheelmap.de...

**Comment by holgerd** [comment]
You are correct, I missed that. I emailed Planetary who is hosting the TLDs for help on this.

**Comment by thermann78** [comment]
So I double-checked wheelmap.de.
In fact there is an A-Record at Planetary redirecting to 176.9.63.170.
Then this request hits Nginx-Conf where it is redirected to wheelmap.org/de - and this is part of the problem as /de doesn't exist, so the autocompletion of WordPress searches for similar URIs leading to a blog article.

In addition, somethin weird happens with accessforeverybody.org: without SSL it redirects to Wheelmap.org. Using https it comes to a completely different site that has nothing to do with us...

**Comment by holgerd** [comment]
^ so I undertand the tasks to fix wheelmap.de are:

- [ ] for Planetary: Change redirect of wheelmap.de from 176.9.63.170 to wheelmap.org
- [ ] for Asquera: remove forwarding rule from nginx (wheelmap.de  to wheelmap.org/de needs to be deleted)

correct?

**Comment by thermann78** [comment]
I think so.

**Comment by Xylakant** [comment]
It's not that simple. wheelmap.org is handled by cloudflare and I think that we can' t just point a different domain there (it's not a redirect if it happens on the DNS level, it's an alias. A redirect requires a webserver to answer a browser request). So we'd still have to point that straight to a webserver we control and handle the redirect there.

It really would be best if we move the external domains to a DNS provider where we have control and can do changes as we need.

**Comment by holgerd** [comment]
@Xylakant Ok. Do you mean something like https://www.inwx.de ?

**Comment by Xylakant** [comment]
INWX is primarily a registrar, that is they handle domains registrations. They do also offer DNS services, but only for domains that are registered with them. we have most of our domains registered at inwx but use dnsimple for DNS services. We don't really need to move the registration to a different registrar, we just need the nameservers somewhere where we can control the dns entries.

**Comment by holgerd** [comment]
@Xylakant & @1000miles We're hosting our domains at [Planetary Networks](https://www.planetary-networks.de/) and have a very good relationship with their customer support.

It features a "cPanel" access with a "simple Zone editor": https://documentation.cpanel.net/display/ALD/Simple+Zone+Editor#

Can we (read: you or somebody else from Asquera) manage it there? The goal is still to redirect the domains I listed in https://github.com/sozialhelden/wheelmap/issues/51#issuecomment-244712636

**Comment by 1000miles** [comment]
@holgerd cc @Svenyo @karolinamakaruk 

@Xylakant and me logged just into the cpanel accounts but we couldn't see all the domains you listed [above](https://github.com/sozialhelden/wheelmap/issues/51#issuecomment-244712636).

**Problem:**
- so far cpanel seems to manage and redirect subdomains only, e.g. `community.wheelmap.org` but not pure domains like `wheelmap.org` *(maybe we are wrong here but we couldn't find any docs about it yet)* => dnsimple.com can do this
- in order to redirect `wheelmap.de` to `wheelmap.org` we need to add it on cloudflare as a separate domain to avoid a `404 - Not Found` but the current cloudflare package doesn't allow to add any other domains that don't belong to wheelmap.org.

Can you please send us the credentials to all cpanel admin interfaces that control the domains (wheelmap.de, wheelmap.com, wheelmap.org, wheelmap.net)? We will have a look into them and will get back to you.

**Comment by holgerd** [comment]
@1000miles the login should be in the 1pw vault now and is called "cPanel accessforeverybody"

**Comment by 1000miles** [comment]
Thank's, @holgerd . We'll have a look into it.

**Comment by 1000miles** [comment]
@holgerd cc @thermann78 @karolinamakaruk 

@Xylakant and me took a look into the cpanel on friday (2016-11-25) and here is our feedback:

- the dns configs of some domains (e.g. wheelmap.de, wheelmap.net, wheelmap.com) in the cpanel don't match the current dns configs 

  => @holgerd Do you know who we can ask where these domains are actually hosted? Maybe the person that set up the configs in the cpanel?

- we still recommend to start using dnsimple in order to manage subdomains & their redirects => please decide here if and when to use dnsimple (or aws, cloudflare, etc. but we recommend dnsimple)

- we could set up SSL/TLS cert via the free [Let's encrypt](https://letsencrypt.org/) (the certificate needs to be updated every 90 days) 

  => @holgerd please let us know if this is required from your side and we can arrange a set up session here

  => **aside note:** the cert in ticket #360 is unfortunately unnecessary at the moment because it is associated to the domain wheelmap.org only and that domain is already covered by cloudflare's SSL/TLS

- short explanation regarding the domain https://accessforeverybody.com: currently the SSL/TLS certificate is on a shared host which is why sometimes it happens that websites are redirected to other websites that use the same cert => solution: host an own SSL/TLS cert

**Comment by Xylakant** [comment]
ok, let me rephrase that for clarity:

The cpanel that we were given to contains configurations that do not match the output if we actually query the name servers. For example, the configuration in the cpanel for wheelmap.com is set to point to `178.23.124.70`, but the nameserver output correctly points to `176.9.63.170`. This indicates that the domains are not actually configured via this cpanel, but via some other configuration that we don't have access to and that we don't know about.

The cpanel also lacks capabitities that we'd need to properly redirect the http requests made by the browser. We'd at least need the capabilities to do redirects via the CPanel, if we want to redirect https requests, we'd either need that capability as well (that would also require a certificate that needs to be provided by or to planetary networks). Or we'd need the capability to add ALIAS records (https://support.dnsimple.com/articles/alias-record/) so that we can point the apex domain to our infrastructure and handle the redirect there.

Therefore we'd propose to at least move the DNS management to a service that offers the relevant capabilities. The domain registration can remain at planetary. There's multiple options to choose from, DNSimple has worked nicely for us in the past. We could also use cloudflare, that would require an account per domain as far as I could see, but on the upside would have SSL built-in. Route 53 would be an option if we already have an AWS account that we can use.

The issue with https://accessforeverybody.com and https://accessforeverybody.org is that both domains point to a shared hosting server which lacks capabilities that allow sharing one IP address for TLS (it's lacking SNI features), so only one domain can have a certificate on that shared server - and that's currently https://ballers-game.de (it's also a self-signed cert, thus the warning popup). So every https connection made to that IP address is made with that domain. There's nothing we can change about that with the current setup. We can fix this either by moving the domain to Cloudflare or by pointing it to our infrastructure and setting up a free Let's Encrypt cert.

The wildcard cert that we bought for the wheelmap.org domain is currently unused because after purchasing the certificate, we decided to move everything under the cloudflare umbrella, and our cloudflare plan does not offer to use our own certificate.

**Comment by 1000miles** [comment]
ping @holgerd :)

**Comment by holgerd** [comment]
@1000miles Yes let's move to dnsimple . Let me know what plan you need.

**Comment by 1000miles** [comment]
@holgerd 

Sounds good. Thank you! :) After consulting @Xylakant we think the smallest plan would be enough for now.

**Comment by holgerd** [comment]
@1000miles dnsimple is setup and you should have access to it via the 1p vault.

**Comment by 1000miles** [comment]
Closing in favor of https://github.com/sozialhelden/wheelmap/issues/510.
